EventsExpress/ClientApp/src/components/event/map-modal.js

```diff
@@ -0,0 +1,144 @@
+﻿import React, { Component } from 'react';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import LocationMap from './map/location-map';
+import { Field } from 'redux-form';
+import eventHelper from '../helpers/eventHelper';
+import './slider.css';
+import DisplayMap from '../event/map/display-map';
+class MapModal extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            open: false,
+            selectedPos: null,
+            radius: this.props.initialValues.radius != undefined ? this.props.initialValues.radius : 0,
+            needInitializeValues: true,
+            map: false
+        }
+        this.baseState = this.state;
+        this.radius = React.createRef();
+    }
+    componentDidUpdate(prevProps) {
+        const initialValues = this.props.initialFormValues;
+        if (!eventHelper.compareObjects(initialValues, prevProps.initialFormValues)
+            || this.state.needInitializeValues) {
+            this.props.initialize({
+                radius: this.props.values.radius,
+                selectedPos: this.props.values.selectedPos
+            });
+            this.setState({
+                ['needInitializeValues']: false
+            });
+        }
+    }
+
+    componentDidUpdate() {
+        return true;
+    }
+
+    handleClickOpen = () => {
+        this.setState({ open: true });
+    };
+    handleClose = () => {
+        this.props.initialize({
+            radius: this.props.initialValues.radius,
+            selectedPos: this.props.initialValues.selectedPos != null ? this.props.initialValues.selectedPos : { lat: null, lng: null }
+        })
+    };
+    handleFilter = () => {
+        this.setState({ open: false });
+    }
+
+    onClickCallBack = (coords) => {
+        this.setState({ selectedPos: [coords.lat, coords.lng] });
+
+    }
+
+    onRadiusChange = (event) => {
+        const { value } = event.target
+        this.setState({ radius: value });
```

Comment:
The value of radius from state is never used. Consider removing this onChange method.